test: set prague as default hardfork

seaona opened the pull request:
<!--
Please submit this PR as a draft initially.
Do not mark it as "Ready for review" until the template has been completely filled out, and PR status checks have passed at least once.
-->

## **Description**

This PR sets `prague` as the default hardfork. This way all our specs will run with the latest hardfork, unless customized otherwise. This implicitly changes all the tx's as type 2 instead of legacy.

For the tests that relied on gas values from a legacy hardfork, we now we need to set the custom hardfork, so they can continue relying on those values. For some tests is obvious they intentionally wanted to use legacy tx's (the test specifies legacy on the description, but for a couple, others, it was not clear, so the corresponding team can decide if they want to migrate -- and update their test values that relied on legacy gas accordingly).

Extra note: the skip e2e quality gate label has been added as this PR shouldn't include new flakiness or race conditions. The spec `[Send ETH from inside MetaMask finds the transaction in the transactions list when sending to a Multisig Address](https://github.com/MetaMask/metamask-extension/blob/main/test/e2e/tests/transaction/send-eth.spec.js) failed 3 time` is a know flaky test, which in this PRis re-run x5 more times, making the pipeline to fail.
This should be tackled in a separate PR as it's an unrelated issue.


[![Open in GitHub Codespaces](https://github.com/codespaces/badge.svg)](https://codespaces.new/MetaMask/metamask-extension/pull/31509?quickstart=1)

## **Related issues**

Fixes: https://github.com/MetaMask/metamask-extension/issues/31757

## **Manual testing steps**
All specs should continue to pass.

## **Screenshots/Recordings**

<!-- If applicable, add screenshots and/or recordings to visualize the before and after of your change. -->

### **Before**

<!-- [screenshots/recordings] -->

### **After**

<!-- [screenshots/recordings] -->

## **Pre-merge author checklist**

- [x] I've followed [MetaMask Contributor Docs](https://github.com/MetaMask/contributor-docs) and [MetaMask Extension Coding Standards](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/CODING_GUIDELINES.md).
- [x] I've completed the PR template to the best of my ability
- [x] I’ve included tests if applicable
- [x] I’ve documented my code using [JSDoc](https://jsdoc.app/) format if applicable
- [x] I’ve applied the right labels on the PR (see [labeling guidelines](https://github.com/MetaMask/metamask-extension/blob/main/.github/guidelines/LABELING_GUIDELINES.md)). Not required for external contributors.

## **Pre-merge reviewer checklist**

- [ ] I've manually tested the PR (e.g. pull and build branch, run the app, test code being changed).
- [ ] I confirm that this PR addresses all acceptance criteria described in the ticket it closes and includes the necessary testing evidence such as recordings and or screenshots.

seaona committed:
set prague as default hardfork

seaona committed:
add legacy tx hardfork

seaona committed:
disable ci filters to test in ff and webpack

seaona committed:
some fixes

seaona committed:
more fixes

seaona committed:
fix remaining failures

seaona committed:
lint

seaona commented on test/e2e/tests/metrics/swaps.spec.js:
```diff
@@ -56,12 +54,6 @@ async function mockSegmentAndMetaswapRequests(mockServer) {
-      .thenCallback(() => ({
-        statusCode: 200,
-        json: GAS_PRICE_API_MOCK_RESULT,
-      })),
```
no gasPrice with type 0x2 tx

seaona committed:
add filters back

seaona committed:
Merge branch 'main' into test-prague-hardfork-default

seaona committed:
Merge branch 'main' into test-prague-hardfork-default

seaona commented on test/e2e/tests/bridge/constants.ts:
```diff
@@ -1,6 +1,6 @@
 import type { FeatureFlagResponse } from '@metamask/bridge-controller';
 
-export const EXPECTED_MAINNET_ETH_BALANCE = '23.9999';
```
balance after tx's is changed slightly as the gas we used before was legacy, now is eip1559

seaona committed:
Merge branch 'main' into test-prague-hardfork-default

seaona committed:
revert

seaona committed:
line

seaona committed:
lint

seaona committed:
fix new/changed testgs with rebased changes

seaona commented on test/e2e/flask/multichain-api/testHelpers.ts:
```diff
@@ -33,19 +33,24 @@ export const DEFAULT_MULTICHAIN_TEST_DAPP_FIXTURE_OPTIONS = {
+        hardfork: 'muirGlacier',
       },
     },
   ],
```
these tests relied on metrics from muirGlacier (ie gas values). Open a separate ticket for updating that as it got convoluted in this same PR -- to discuss with multichain team

seaona committed:
alst tweaks

seaona commented on test/e2e/helpers.js:
```diff
@@ -544,11 +543,6 @@ const PRIVATE_KEY_TWO =
-  // EVM version that supports type 2 transactions (EIP1559)
-  hardfork: 'london',
-};
-
```
eip1559 is now supported by default, as we use prague

seaona committed:
Merge branch 'main' into test-prague-hardfork-default

seaona commented on test/e2e/tests/transaction/send-eth.spec.js:
```diff
@@ -157,14 +155,6 @@ describe('Send ETH', function () {
-                hardfork: 'london',
-              },
-            },
-          ],
```
we can remove all the cutstom options as these was done initially when anvil was not run by default in all specs. So now, we have default anvil and default prague

hjetpoluru commented on test/e2e/tests/bridge/constants.ts:
```diff
@@ -1,5 +1,7 @@
 import type { FeatureFlagResponse } from '@metamask/bridge-controller';
 
+export const EXPECTED_MAINNET_ETH_BALANCE = '23.9977';
```
This variable is not used anywhere is this expected?

seaona replied:
whoops! Nice catch :heart_eyes: , not expected! It's a leftover from another change I had local. Removing this now :pray: ty!

seaona committed:
remov unused var - review comment from harika

hjetpoluru reviewed (dismissed):
LGTM

cortisiko reviewed (dismissed):
🌮 🌮

seaona committed:
Merge branch 'main' into test-prague-hardfork-default

HowardBraham commented:
@seaona @cortisiko @GustavoRSSilva someone made a merge mistake here in `test/e2e/tests/bridge/bridge-test-utils.ts` and it's failing lint

seaona committed:
lint fix

seaona committed:
fix new test

seaona committed:
fix

seaona commented:
> @seaona @cortisiko @GustavoRSSilva someone made a merge mistake here in test/e2e/tests/bridge/bridge-test-utils.ts and it's failing lint

thank you vmuch for following up on this one @HowardBraham :heart_eyes: I've just rebased and added a couple of fixes now. The PR is ready again for re-review
